### PR TITLE
[CLU] Edited ReadMe.md 

### PR DESCRIPTION
**Pull request by AhmedLeithy**
Adjusted Readme. However the references (links) to service documentation should be adjusted in a later PR; this is because CLU docs are currently in private preview and not yet available to the public.

Also one thing to note: In the general troubleshooting section, line 97, it is stated that this invalid request should return a 400 (bad request error). However, when I tested it myself (using my own resource), I got a 401 unauthorized access error, although both the key and the endpoint TestEnvironment variables were correctly set. Could this be a service issue? Can someone please try to run that test using their resources and let me know if they get the correct 400 exception?

### CODE REVIEW

**Review by heaths — CHANGES_REQUESTED** [review]
Let's get the initial client in first and we can rebase this one.

**Comment by AhmedLeithy** [comment]
We can add more samples to the readme (using options etc.) once they're completed (Salah is currently working on them). However, the current ReadMe has been rebased.

**Comment by heaths** [comment]
1 or 2 more samples for common scenarios in the README would be good, but let's keep it only to common things users might do, i.e. the "hero scenarios" we have in our other READMEs. Additional, less-common tasks can be put into dedicated samples you'll find in some other packages e.g. sdk/keyvault/Azure.Security.KeyVault.Secrets/samples (these use snippets from the sibling test/samples directory).

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:3]: "LuisVNext" not only seems formatted incorrect ("LUIS vNext"?) but a term customers probably aren't familiar with nor would they need to know. Maybe instead something like, "Azure Conversations - the new version of Language Understanding (LUIS) - is a cloud-based ..."

Should also say "pulls out ..." for word agreement with "applies custom ...".

```suggestion
Azure Conversations - the new version of Language Understanding (LUIS) - is a cloud-based conversational AI service that applies custom machine-learning intelligence to a user's conversational, natural language text to predict overall meaning; and pulls out relevant, detailed information. The service utilizes state-of-the-art technology to create and utilize natively multilingual models, which means that users would be able to train their models in one language but predict in others. 
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:20]: Is "Text Analytics" here right? Is it the same resource type? At the very least, we may want to point that out, but I believe they are different resource types.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:42]: These code changes actually need to be made in the tests/Azure.AI.Language.Conversations.Tests/samples directory, then you run (rrom the repo root) `eng/scripts/CodeChecks.ps1 cognitivelanguage` to update them and make sure everything is correct. That's why this PR build is also failing.

Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:42]: We do this to make sure 1) our samples actually compile, and 2) when recorded or run live, the samples actually work like we expect.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:53]: Nit (consistency): 
```suggestion
The [`ConversationAnalysisClient`][conversationanalysis_client_class] is the primary interface for making predictions using your deployed Conversations models. It provides both synchronous and asynchronous APIs to submit queries.
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:77]:
```suggestion
### Analyze a conversation
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:79]:
```suggestion
To analyze a conversation, we can then call the `client.AnalyzeConversation()` method which takes the project name, deployment name, and query as parameters.
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:42]: ...or are these from my previous changes?

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:90]:
```suggestion
To set additional properties such as language of the query and verbosity, you can initialize an `AnalyzeConversationOptions` instance with your chosen parameters. You can then call `AnalyzeConversation()` using the options as a parameter.
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:142]: This should be kept. It's standard in all our SDK READMEs.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:156]: Comment is unnecessary (and misspelled). Our linker checker takes into account if a PR's links *will* be in main after merging.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:164]: Let's open tracking issues for these. Be sure to add them to the "Cognitive Services - Language" label.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversation.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversation.md:1]:
```suggestion
# Analyze a conversation
```

**Review by heaths — CHANGES_REQUESTED** [review]
Most a few nits.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (AhmedLeithy) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:142]: The original message is about the "azure search service". I wanted to recreate the error for the conversations service using the `BadRequest` Test in Readme.cs. But I received a 401 error instead. I will bring back the original message.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (AhmedLeithy) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:20]: When I use the CLU Language Studio (the one in preview) to create new projects, I need to choose an Azure language resource of type "Text Analytics". I am not sure if this will still be the case after preview is over, I'll check and let you know.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (AhmedLeithy) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:185]: Unfortunately, I can't add any labels to issues :(. Does this also have to do with the write access that I don't have?

Here is a link to the issue. Can you please add the label ? [https://github.com/Azure/azure-sdk-for-net/issues/24163]( https://github.com/Azure/azure-sdk-for-net/issues/24163)

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (AhmedLeithy) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md:20]: I checked with a service team privately, and it seems like this will remain to be the case.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs:9]: Why was this moved out of "Generated"? In general, we don't want to take manual ownership of these files.

Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs:9]: I also don't see any significant difference - just whitespace changes.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversationWithOptions.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversationWithOptions.cs:1]: There already is a Sample1. You need to do Sample2. The name after "_" doesn't matter.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversationWithOptions.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversationWithOptions.cs:23]: If you're going to split parameters across lines, do it for all parameters.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample2_AnalyzeConversationWithLanguage.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample2_AnalyzeConversationWithLanguage.cs:1]: "Sample3_..."

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample3_Deepstack.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample3_Deepstack.cs:1]: "Sample4_..."

**Review by heaths — CHANGES_REQUESTED** [review]
All those samples also need corresponding markdown files in sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples. See that folder for an example.

**Comment by heaths** [comment]
See https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples. You should run `eng/scripts/CodeChecks.ps1 cognitivelanguage` before pushing to find any errors more quickly. You also need to record tests per the offline instructions I gave you, hence the CI build errors.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs
Comment (ZulaMostafa) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs:9]: Yeah I wanted to tell you about this one. There is a mismatch between the swagger and the actual response in the field named `projectType`. it's name in the swagger is `projectKind` while the actual response has `projectType`. So I had to change it based on the [renaming guidelines](https://github.com/chamons/autorest.csharp#customize-serializationdeserialization-methods)

**Comment by ZulaMostafa** [comment]
There are actually markdown files for all the samples. But need to rename them alongside the samples.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversationWithOptions.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversationWithOptions.md:19]: When putting parameters on separate lines, lets do so consistently with all of them on separate lines like examples below. It can help with readability, especially with new people to the library when the parameter values are long.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversationWithOptions.md
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/samples/Sample1_AnalyzeConversationWithOptions.md:1]: There's already a ["Sample1_*" set of files](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs), so this - and subsequence samples, should be offset +1 e.g. Sample2_, etc. You might still keep them in the same order you have them, though. They are designed to be progressively harder / more advanced.

**Review by heaths — APPROVED** [review]
Just a nit or two, but the builds are still failing. Be sure to click on "Details" and find out why. And certainly feel free to ask questions if it's not clear.

**Comment by heaths** [comment]
Most, if not all (I didn't check all of them) of the failures are because you didn't (re) record the tests e.g. 

```
Azure.Core.TestFramework.TestRecordingMismatchException : Could not find file 'D:\a_work\1\s\sdk\cognitivelanguage\Azure.AI.Language.Conversations\tests\SessionRecords\ConversationAnalysisClientSamples\AnalyzeConversation.json'.
----> System.IO.FileNotFoundException : Could not find file 'D:\a_work\1\s\sdk\cognitivelanguage\Azure.AI.Language.Conversations\tests\SessionRecords\ConversationAnalysisClientSamples\AnalyzeConversation.json'.
```

Until we're able to deploy the resources, you need to set environment variables manually:

Create a service principal: `az ad sp create-for-rbac

Set environment variables for the appId, password, and tenant it returns. I recommend also using setx on Windows to remember these for future sessions, or just use the Environment variables control panel.
```cmd
set AZURE_CLIENT_ID=<appId>
set AZURE_CLIENT_SECRET=<password>
set AZURE_TENANT_ID=<tenant>
```
You may also need to set the `AZURE_SUBSCRIPTION_ID` to whatever sub contains those resources

Set the test-specific environment variables:
```cmd
set CONVERSATIONS_KEYS=<key I gave you previously>
set CONVERSATIONS_PROJECT=<projectName>
set CONVERSATIONS_URI=<PPE URI>
```
Finally, you either need to set an environment variable named `AZURE_TEST_MODE=Record` or, inside of Visual Studio, find the test Base class and make sure the last parameter in the constructor is set to `RecordedTestMode.Record`.

Run your tests and that will create some JSON files under tests/SessionRecords. Review them brief to make sure they look right. Make sure you assert conditions that are consistent as well (with ML this can be hard). Before checking in, make sure you revert the constructor parameter back.

**Comment by heaths** [comment]
If it would help, we could do a walkthrough on Teams.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Infrastructure/ConversationAnalysisClientWithLanguageSamples.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Infrastructure/ConversationAnalysisClientWithLanguageSamples.cs:1]: Was there a reason for this base class? It doesn't seem to add any capabilities that are useful for derivative classes.

Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Infrastructure/ConversationAnalysisClientWithLanguageSamples.cs:1]: You can (and probably should) just use the same partial class in different files and add methods. Partial classes can be spread across many, many files.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs:37]: Why remove these? You should assert something - ideally, something from the expected response itself, though I can understand that can be difficult in cases like this.

**Review by heaths — CHANGES_REQUESTED** [review]
Almost there. Open question on why new classes were created. It shouldn't be necessary and is less to maintain to just use a single base class.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs
Comment (ZulaMostafa) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs:37]: I think we should be asserting in tests instead of samples, that's why I removed it. I've also made some tests with asserting the answer but haven't pushed them yet. Thought it would be better if I made a separate PR after merging this one.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs
Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/BasePrediction.cs:9]: Taking over the de/serialization, though, is very risky. Instead, in this case, you can use a transform. Adding the following to autorest.md would be best, and make sure to file a bug assigned to @ChongTang so he knows.

In autorest.md, look for the "transform" block and, at the bottom of the first one, add:

```yaml
- from: swagger-document
  where: $.definitions.BasePrediction
  transform: |
    $.discriminator = "projectType";
    $.required = [ "projectType" ];
    $.projectType = $.projectKind;
    delete $.projectKind;
```

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md
Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/README.md]: You could generate an error and copy that out.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs
Comment (heaths) [reply sdk/cognitivelanguage/Azure.AI.Language.Conversations/tests/Samples/Sample1_AnalyzeConversation.cs:37]: Samples are tests, and designed as such. We want to make sure the test compile and actually work. Only the code between the regions will be extract. Please add the asserts back in.

**Review by heaths — APPROVED** [review]

**Pull request merged**
